packages/core/content-type-builder/admin/src/components/AIChat/hooks/useAIFetch.ts

```diff
@@ -150,35 +254,142 @@ export const createAIFetchHook = <T extends keyof AIEndpoints>(endpoint: T) => {
+        const fallbackToken = window.strapi?.aiLicenseKey;
+        if (fallbackToken) {
+          return fallbackToken;
+        }
+
+        // 2. Check for stored valid token
+        const storedToken = getStoredAIToken();
+        if (storedToken) {
+          return storedToken.token;
+        }
+
+        // 3. Fetch new token
+        const getAdminToken = () => {
+          const STORAGE_KEYS = { TOKEN: 'jwtToken' };
+
+          const fromLocalStorage = localStorage.getItem(STORAGE_KEYS.TOKEN);
+
+          if (fromLocalStorage) {
+            try {
+              const parsed = JSON.parse(fromLocalStorage);
+              return parsed;
+            } catch (e) {
+              return fromLocalStorage;
+            }
+          }
+
+          // Also try cookies as fallback
+          const fromCookie = document.cookie
+            .split(';')
+            .find((row) => row.trim().startsWith('jwtToken='));
+
+          return fromCookie ? fromCookie.split('=')[1] : null;
+        };
+
+        const adminToken = getAdminToken();
+
+        if (!adminToken) {
+          return null;
+        }
+
+        const response = await fetch('/admin/users/me/ai-token', {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${adminToken}`,
+          },
+        });
+
+        if (!response.ok) {
+          await response.text();
+          return null;
+        }
+
+        const data = await response.json();
+
+        const token = data?.token || data?.data?.token;
+        const expiresAt = data?.expiresAt || data?.data?.expiresAt;
+
+        // Store the token for future use
+        if (token && expiresAt) {
+          storeAIToken({ token, expiresAt });
+        }
+
+        return token || null;
+      } catch (error) {
+        return null;
+      }
+    };
+
+    /**
+     * Make a type-safe API call to the specified Strapi AI endpoint with retry logic
      */
     const fetchData = async (
-      options: Omit<RequestInit, 'body'> & Partial<RequestType<T>> & { formData?: FormData } = {}
+      options: Omit<RequestInit, 'body'> & Partial<RequestType<T>> & { formData?: FormData } = {},
+      isRetry = false
     ): Promise<ResponseType<T> | null> => {
       setIsPending(true);
       setError(null);
 
       try {
-        const headers = {
-          Authorization: `Bearer ${STRAPI_AI_TOKEN}`,
-          'X-Strapi-Version': strapiVersion || 'latest',
-          'X-Strapi-User': userId || 'unknown',
-          'X-Strapi-Project-Id': projectId || 'unknown',
-          ...options.headers,
-        } as Record<string, string>;
-
-        if (options.body) {
-          headers['Content-Type'] = 'application/json';
+        // Get AI token
+        const aiToken = await getAIToken();
+        if (!aiToken) {
+          setError('No AI token available. Please ensure you have Enterprise Edition license.');
```

Comment:
```suggestion
          setError('Could not authorize with AI Server. Please contact your adminstrator.');
```
@maccomaccomaccomacco you may want to review all the error messages in here to make sure they align with what you expect